auth/credentials/internal/impersonate/impersonate.go

```diff
@@ -81,6 +83,13 @@ type Options struct {
 	// enabled by setting GOOGLE_SDK_GO_LOGGING_LEVEL in which case a default
 	// logger will be used. Optional.
 	Logger *slog.Logger
+	// TrustBoundaryDataProvider is used to fetch trust boundary data.
+	// This data defines the regions or environments where the credential (and subsequently the tokens
+	// obtained by it) is allowed to be used, enforcing trust boundary restrictions.
+	// If nil, no trust boundary restrictions are applied or fetched for this flow.
+	TrustBoundaryDataProvider trustboundary.DataProvider
+	// UniverseDomain is the default service domain for a given Cloud universe.
+	UniverseDomain string
```

Comment:
@quartzmo Was this missed in previous universe domain work, or are we handling UD things differently for impersonated creds?

Reply:
Do you mean, why is it a string and not a provider? It's because there should be no need to retrieve the value from a late-binding source such as MDS.

Reply:
This Options type is for NewTokenProvider, and so far we haven't supported UniverseDomain in the TokenProvider interface, only in Credentials.